### ISSUE TITLE
Black screen with Raspbian's kernel8.img

### ISSUE THREAD

**Issue by alkisg** [ipxe/pipxe#1]
Thanks for pipxe, it sounds awesome!
It booted on the first try, but:

- It takes 3 minutes to load the kernel/initrd via TFTP, is http faster?
- The kernel8.img from the latest Raspbian Buster just shows a black screen after it loads. The other Raspbian kernels show Exec format error.

Which kernel is appropriate to boot an rpi3 with pipxe?

**Comment by alkisg** [comment]
Testing with https://github.com/Crazyhead90/pi64/releases and specifically [kernel8.img from there](https://github.com/Crazyhead90/pi64/releases/download/2018-04-17/Bamarni-lite-full64bit-pi3bplus-V2.zip) did allow the kernel to write on screen even though it panicked later as I didn't use an initramfs.

So currently there's no way to boot Raspbian with pipxe?

**Comment by alkisg** [comment]
On IRC, mcb30 said that [this patch of his](https://github.com/raspberrypi/linux/pull/3252) is needed. This isn't yet available on Raspbian, so I had to use a newer firmware from https://github.com/Hexxeh/rpi-firmware.

Now Raspbian netboots almost fine, with that 64bit kernel and the usual 32bit userland.

One problem is that Debian's ipconfig doesn't wait long enough for udev to make the NIC available. I'll file that against Debian.

A second problem that might be related to the kernel patch above though, is that the colors are wrong; I think it swaps blue with green or something.

Closing from here though as pipxe does its work fine. Thanks mcb30!
